### ISSUE TITLE
Index gets stuck in installed state if a transaction is open

### ISSUE THREAD

**Issue by BrynCooke** [thinkaurelius/titan#1051]
If the user has an open transaction then new index will never make it to installed state

```
g = TitanFactory.open('conf/titan-cassandra-embedded-es.properties')
GraphOfTheGodsFactory.load(g)
g.traversal().V().has('name', CONTAINS, 'hercules').order().by('age', decr).limit(10)

//g.tx().rollback(); <--*** uncomment to make things work ***

mgmt = g.openManagement()
name = mgmt.getPropertyKey('name')
age = mgmt.getPropertyKey('age')
mgmt.buildIndex('nameAndAge', Vertex.class).addKey(name).addKey(age).buildMixedIndex("search")
mgmt.commit()
//Wait for the index to become available
mgmt.awaitGraphIndexStatus(g, 'nameAndAge').call() //Never happens

```

**Comment by mbroecheler** [comment]
Yes, this is unfortunately a feature and not a bug. We cannot install an index while a transaction is still open, because that transaction might interact with the database using the old index state (since that information is cached and cannot be changed mid-transaction). Hence, we need to wait for all tx to close before we can signal the update.
@BrynCooke Please point this out in the documentation (with emphasis). In addition, I created #1059 which could be a remedy.

**Comment by dpitera** [comment]
Is this also true for reindexing indexes? Can I concurrently `reindex` a `registered` indexes, or will these indexes also get stuck in the `registered` state?

**Comment by dpitera** [comment]
> We cannot install an index while a transaction is still open, because that transaction might interact with the database using the old index state (since that information is cached and cannot be changed mid-transaction). Hence, we need to wait for all tx to close before we can signal the update.

Can you point out where exactly in the code while installing an index we check for open transactions and handle accordinly @mbroecheler

**Comment by ngrigoriev** [comment]
I was able to reliably reproduce this problem on Titan 1.0.0. Here is how:

1. Start one Gremlin server and connect Gremlin console to it
2. Kill that server (-9) so it cannot shut itself down properly
3. Start another Gremlin server
4. Try to create an index like this:

(snippet 1)
```
		final String indexName = "mixedIndex12";
		final PropertyKey pk1 = mgmt.getPropertyKey("some_property");
		final PropertyKey pk2 = mgmt.getPropertyKey("another_property");
		mgmt.buildIndex(indexName, Vertex.class).addKey(pk1).addKey(pk2).buildMixedIndex("search");
		mgmt.commit();

		ManagementSystem.awaitGraphIndexStatus(graph, indexName).call();
```

You can try to register that index as many times as you want, this will never work, it will get stuck and time out:

(snippet 2)

```
		mgmt.updateIndex(mgmt.getGraphIndex(indexName), SchemaAction.REGISTER_INDEX).get();
		mgmt.commit();

		ManagementSystem.awaitGraphIndexStatus(graph, indexName).call();
```

My original thought was that it was because of the stale instance (the one I have kill -9'ed). Removing this instance using mgmt.forceCloseInstance(...) does not help. 

5. Remove the stale instance. You can use Gremlin console:

(snippet 3)

```
:> mgmt = graph.openManagement() ; mgmt.getOpenInstances();
:> mgmt = graph.openManagement(); mgmt.forceCloseInstance("0a67c83e17345-myhost-local1"); mgmt.commit();
```

6. Repeat the snippet 2. It still won't work, it will time out. You will see the messages logged by Gremlin server about stale transactions.

7. Shut down your Gremlin server, see step #3. You can start it again or keep it down, does not matter.

8. Try snippet 2 again. It will work, you will get your index registered.

So, it seems that removing the stale instance leaves something else that gets cleaned up by shutting down the active Titan servers.
